File: backend/app/schemas/family.py
from datetime import datetime

from pydantic import BaseModel


class FamilyCreate(BaseModel):
    name: str


class FamilyJoin(BaseModel):
    invite_code: str


class FamilyResponse(BaseModel):
    id: int
    name: str
    invite_code: str
    created_at: datetime
    default_family_calendar_category_id: int | None = None

    model_config = {"from_attributes": True}


class FamilySettingsUpdate(BaseModel):
    default_family_calendar_category_id: int | None = None

File: obsolete/ios/Familienkalender/Models/Family.swift
import Foundation

struct FamilyCreate: Codable {
    let name: String
}

struct FamilyJoin: Codable {
    let inviteCode: String
}

struct FamilyResponse: Codable, Identifiable {
    let id: Int
    let name: String
    let inviteCode: String
    let createdAt: String
}
